### ISSUE TITLE
Broken hooks

### ISSUE THREAD

**Issue by jValdron** [sequelize/sequelize#1324]
Since 1.7.0-rc1, when the hooks got changed to DAO instances instead of just the values hash, my beforeUpdate hook doesn't seem to work as it used to.

I have a really simple hook which looked something like this:

``` javascript
beforeUpdate: function(values, cb){

    this.find(values.id).success(function(original){

        Object.keys(values).forEach(function(key){
            if (key.indexOf('_need_work') > -1 && key.indexOf('_started_on') === -1 && values[key] === 'yes' && (original[key] === 'no' || !original[key])) {
                values[key + '_started_on'] = new Date();
            }
        });

        return cb(null, values);

    });

}
```

Essentially, it just puts a timestamp in fields if another ENUM field has been switch to 'yes'.

I changed this to use a DAO object, so intead of using Object.keys(values) I do values.dataValues (just want to loop though all the fields in my object, not that optimal). And I also removed the values in the callback.

I checked my object with console.log, the timestamp was updated and everything, however, when they query gets executed, I can see in the query that the timestamp doesn't get updated.

Long story short, is there anything special that we have to do in other to update values of our DAO instances in the hooks? Do we have to flag the field as dirty?
